File: src/main/java/alfheim/common/potion/PotionSoulburn.java
package alfheim.common.potion;

import org.lwjgl.opengl.GL11;

import alfheim.common.block.BlockRedFlame;
import alfheim.common.core.registry.AlfheimBlocks;
import alfheim.common.core.util.AlfheimConfig;
import alfheim.common.core.util.DamageSourceSpell;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.client.Minecraft;
import net.minecraft.client.renderer.OpenGlHelper;
import net.minecraft.client.renderer.Tessellator;
import net.minecraft.client.renderer.entity.Render;
import net.minecraft.client.renderer.entity.RenderManager;
import net.minecraft.client.renderer.texture.TextureMap;
import net.minecraft.entity.Entity;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.util.IIcon;

public class PotionSoulburn extends PotionAlfheim {

	public PotionSoulburn() {
		super(AlfheimConfig.potionIDSoulburn, "soulburn", true, 0xCC4400);
	}

	@Override
	public boolean isReady(int time, int mod) {
		return time % 20 == 0; 
	}

	@Override
	public void performEffect(EntityLivingBase living, int mod) {
		living.attackEntityFrom(DamageSourceSpell.soulburn, mod+1);
	}
	
	@SideOnly(Side.CLIENT)
	public static void renderFireInFirstPerson(float partialTicks) {
		Tessellator tessellator = Tessellator.instance;
		GL11.glColor4f(1.0F, 1.0F, 1.0F, 0.9F);
		GL11.glEnable(GL11.GL_BLEND);
		OpenGlHelper.glBlendFunc(770, 771, 1, 0);
		float f1 = 1.0F;

		for (int i = 0; i < 2; ++i) {
			GL11.glPushMatrix();
			IIcon iicon = ((BlockRedFlame) AlfheimBlocks.redFlame).icons[1];
			Minecraft.getMinecraft().getTextureManager().bindTexture(TextureMap.locationBlocksTexture);
			float f2 = iicon.getMinU();
			float f3 = iicon.getMaxU();
			float f4 = iicon.getMinV();
			float f5 = iicon.getMaxV();
			float f6 = (0.0F - f1) / 2.0F;
			float f7 = f6 + f1;
			float f8 = 0.0F - f1 / 2.0F;
			float f9 = f8 + f1;
			float f10 = -0.5F;
			GL11.glTranslatef((float)(-(i * 2 - 1)) * 0.24F, -0.3F, 0.0F);
			GL11.glRotatef((float)(i * 2 - 1) * 10.0F, 0.0F, 1.0F, 0.0F);
			tessellator.startDrawingQuads();
			tessellator.addVertexWithUV((double)f6, (double)f8, (double)f10, (double)f3, (double)f5);
			tessellator.addVertexWithUV((double)f7, (double)f8, (double)f10, (double)f2, (double)f5);
			tessellator.addVertexWithUV((double)f7, (double)f9, (double)f10, (double)f2, (double)f4);
			tessellator.addVertexWithUV((double)f6, (double)f9, (double)f10, (double)f3, (double)f4);
			tessellator.draw();
			GL11.glPopMatrix();
		}

		GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
		GL11.glDisable(GL11.GL_BLEND);
	}
}

File: src/main/java/alfheim/common/potion/PotionSoulburn.kt
package alfheim.common.potion

import alexsocol.asjlib.*
import alfheim.common.block.*
import alfheim.common.core.handler.AlfheimConfigHandler
import alfheim.common.core.util.DamageSourceSpell
import cpw.mods.fml.relauncher.*
import net.minecraft.client.renderer.*
import net.minecraft.client.renderer.texture.TextureMap
import net.minecraft.entity.EntityLivingBase
import org.lwjgl.opengl.GL11.*

class PotionSoulburn: PotionAlfheim(AlfheimConfigHandler.potionIDSoulburn, "soulburn", true, 0xCC4400) {
	
	override fun isReady(time: Int, mod: Int): Boolean {
		return time % 20 == 0
	}
	
	override fun performEffect(living: EntityLivingBase, mod: Int) {
		living.attackEntityFrom(DamageSourceSpell.soulburn, (mod + 1).F)
	}
	
	companion object {
		
		@SideOnly(Side.CLIENT)
		fun renderFireInFirstPerson(partialTicks: Float) {
			val tessellator = Tessellator.instance
			glColor4f(1f, 1f, 1f, 0.9f)
			glEnable(GL_BLEND)
			OpenGlHelper.glBlendFunc(770, 771, 1, 0)
			val f1 = 1f
			
			for (i in 0..1) {
				glPushMatrix()
				val iicon = (AlfheimBlocks.redFlame as BlockRedFlame).icons[1]
				mc.textureManager.bindTexture(TextureMap.locationBlocksTexture)
				val f2 = iicon.minU
				val f3 = iicon.maxU
				val f4 = iicon.minV
				val f5 = iicon.maxV
				val f6 = (0f - f1) / 2f
				val f7 = f6 + f1
				val f8 = 0f - f1 / 2f
				val f9 = f8 + f1
				val f10 = -0.5f
				glTranslatef((-(i * 2 - 1)).F * 0.24f, -0.3f, 0f)
				glRotatef((i * 2 - 1).F * 10f, 0f, 1f, 0f)
				tessellator.startDrawingQuads()
				tessellator.addVertexWithUV(f6.D, f8.D, f10.D, f3.D, f5.D)
				tessellator.addVertexWithUV(f7.D, f8.D, f10.D, f2.D, f5.D)
				tessellator.addVertexWithUV(f7.D, f9.D, f10.D, f2.D, f4.D)
				tessellator.addVertexWithUV(f6.D, f9.D, f10.D, f3.D, f4.D)
				tessellator.draw()
				glPopMatrix()
			}
			
			glColor4f(1f, 1f, 1f, 1f)
			glDisable(GL_BLEND)
		}
	}
}
